### ISSUE TITLE
How to visualize intermediate layer output that are learnt during training process?

### ISSUE THREAD

**Issue by ChidanandKumarKS** [apache/incubator-mxnet#7229]
For bugs or installation issues, please provide the following information.
The more information you provide, the more likely people will be able to help you.

## Environment info
Operating System: Ubuntu 16.0.4

Compiler: python 2.7

Package used (Python/R/Scala/Julia):Python

MXNet version:0.9.3

Or if installed from source:

MXNet commit hash (`git rev-parse HEAD`):

If you are using python package, please provide

Python version and distribution:2.7.13

How to visualize intermediate layer output that are learnt during training process? Any solution

**Comment by szha** [comment]
[monitor](https://github.com/dmlc/mxnet/blob/b3b21e4e2006a54ea11573f95d439b6d5d8aa928/python/mxnet/monitor.py#L16) can be [installed](http://mxnet.io/api/python/module.html#mxnet.module.BaseModule.install_monitor) when training in symbolic API with modules, assuming you're asking about symbolic API. Also, you may consider adopting [Gluon](http://mxnet.io/api/python/gluon.html], which provides imperative mode and layer abstractions that are easier to work with.

**Comment by ChidanandKumarKS** [comment]
I have gone thru docs. I have seen "monitor"  is often helpful to print out the weights and intermediate outputs to understand what's going on and accuracy.
But i would like to visualize the outputs in terms of images for every layers(each layers has 2D several feature maps). I would like to visualize 2D feature maps in terms of images for every layer. Is there is such alternative.

**Comment by M4xim4l** [comment]
I fully agree, having something like Keras-vis (https://raghakot.github.io/keras-vis/#getting-started) for mxnet would be super cool, but I don't think there is any code available for that, at least I haven't found any. But it definitely should be possible to implement something like guided backprop for mxnet.

**Comment by szha** [comment]
@zihaolucky has been working on an independent version of tensorboard that can serve this purpose.

**Comment by eric-haibin-lin** [comment]
@zihaolucky is the tensorboard-lite ready?

**Comment by sandeep-krishnamurthy** [comment]
MXBoard should solve this use case - https://github.com/awslabs/mxboard
